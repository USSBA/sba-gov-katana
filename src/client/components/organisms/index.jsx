/* IMPORTS */

// miscellaenous organisms
import BusinessGuideTileCollection from "./business-guide-tile-collection/business-guide-tile-collection.jsx";
import CardCollection from "./card-collection/card-collection.jsx";
import DeveloperTester from "./developer-tester/developer-tester.jsx";
import DocumentCardCollection from "./document-card-collection/document-card-collection.jsx";
import FrontPageHero from "./front-page-hero/front-page-hero.jsx";
import FundingProgramsTileCollection from "./funding-programs-tile-collection/funding-programs-tile-collection.jsx";
import DisasterAlert from "./header-footer/disaster-alert/disaster-alert.jsx";
import NotificationBar from "./header-footer/notification-bar.jsx";
import Hero from "./hero/hero.jsx";
import RelatedDocumentCards from "./related-document-cards/related-document-cards.jsx";
import SearchBox from "./search-box/search-box.jsx";
import SectionNav from "./section-nav/section-nav.jsx";

// header-footer organisms
import Footer from "./header-footer/footer/footer.jsx";
import MobileNav from "./header-footer/mobile-nav/mobile-nav.jsx";
import Header from "./header-footer/header/header.jsx";

// homepage organisms
import Blog from "./homepage/blog/blog.jsx";
import FrontPageLady from "./homepage/front-page-lady/front-page-lady.jsx";
import HappeningNow from "./homepage/happening-now/happening-now.jsx";

// lender match organisms
// -- form
import AdditionalInfoForm from "./lender-match/form/additional-info.jsx";
import BusinessInfoForm from "./lender-match/form/business-info.jsx";
import ContactInfoForm from "./lender-match/form/contact-info.jsx";
import IndustryInfoForm from "./lender-match/form/industry-info.jsx";
import LoanInfo from "./lender-match/form/loan-info.jsx";
import ReviewSection from "./lender-match/form/review-page-helpers.jsx";
import ReviewSubmitInfoForm from "./lender-match/form/review-submit-info.jsx";
// -- landing page
import FindLendersIntro from "./lender-match/landing-page/find-lenders-intro.jsx";
import HelpfulQuestions from "./lender-match/landing-page/helpful-questions.jsx";
import {HowItWorksSection} from "./lender-match/landing-page/how-it-works.jsx";
import PreparationChecklist from "./lender-match/landing-page/preparation-checklist.jsx";
// -- success page
import ConfirmSection from "./lender-match/success-page/confirm-section.jsx";
import DynamicCounselingAndTools from "./lender-match/success-page/counseling-and-tools.jsx";
import CounselorMap from "./lender-match/success-page/counselor-map.jsx";
import EmailConfirmationInvalid from "./lender-match/success-page/email-confirmed-page.jsx";

// modal organisms
import LeaveSbaModal from "./modals/leave-sba-modal/leave-sba-modal.jsx";
import MobileSectionNavModal from "./modals/mobile-section-nav/mobile-section-nav.jsx";

/* EXPORTS */

// miscellaneous organisms
export {
	BusinessGuideTileCollection,
	CardCollection,
	DeveloperTester,
	DocumentCardCollection,
	FrontPageHero,
	FundingProgramsTileCollection,
	DisasterAlert,
	NotificationBar,
	Hero,
	RelatedDocumentCards,
	SearchBox,
	SectionNav
};

// header-footer organisms
export {
	Footer,
	MobileNav,
	Header
};

// homepage organisms
export {
	Blog,
	FrontPageLady,
	HappeningNow
};

// lender match organisms
export {
	AdditionalInfoForm,
	BusinessInfoForm,
	ContactInfoForm,
	IndustryInfoForm,
	LoanInfo,
	ReviewSection,
	ReviewSubmitInfoForm,
	FindLendersIntro,
	HelpfulQuestions,
	HowItWorksSection,
	PreparationChecklist,
	ConfirmSection,
	DynamicCounselingAndTools,
	CounselorMap,
	EmailConfirmationInvalid
};

// modal organisms
export {
	LeaveSbaModal,
	MobileSectionNavModal
};
