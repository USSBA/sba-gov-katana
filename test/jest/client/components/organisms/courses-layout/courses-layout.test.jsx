import React from 'react'
import renderer from 'react-test-renderer'

import { CoursesLayout } from 'organisms'

describe('CoursesLayout', () => {
  test('should render the first page properly', () => {
    const TOTAL = 25

    const mockCourse = {
      body: '<p>some html</p>',
      courseCategory: ['Grow your business'],
      courseMaterials: [
        {
          url: '/sites/default/files/2018-02/ANC%20Workbook%20P.pdf',
          description: 'Workbook'
        }
      ],
      courseType: 'Flash',
      courseUrl: {
        url: 'https://s3.amazonaws.com/training.sba.gov/courses/2013/anc/index.htm',
        title: ''
      },
      image: {
        url: '/sites/default/files/2018-02/shutterstock_649579099_500.jpg',
        alt: 'ANC Business Guide: 8(a) Business Development Program'
      },
      summary:
        'Overview of assistance available to Alaska Native Corporations from the 8(a) Business Development Program.',
      transcript: '<p>some html</p>',
      type: 'course',
      title: 'ANC Business Guide: 8(a) Business Development Program',
      id: 5922,
      updated: 1518720798,
      created: 1517955056,
      url: '/course/anc-business-guide-8a-business-development-program'
    }
    const mockCourses = Array.apply(null, Array(TOTAL)).map((_, i) => mockCourse)

    const component = renderer.create(<CoursesLayout items={mockCourses} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
