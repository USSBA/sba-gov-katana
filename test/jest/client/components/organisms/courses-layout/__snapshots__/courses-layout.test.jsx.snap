// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoursesLayout should render the first page properly 1`] = `
<div
  className="courses-layout container"
>
  <div
    className="client-paging-multiview-layout container"
  >
    <div>
      <div />
      <div
        className="paginator"
      >
        <div
          className="container paginator"
          id={undefined}
        >
          <p
            className="pagination-status"
          >
            <span>
              Showing  
            </span>
            <span
              className="bold"
            >
              1
                - 
              12
                
            </span>
            <span>
              of  
            </span>
            <span
              className="bold"
            >
              25
            </span>
          </p>
          <span
            className="previous-arrow"
          >
            <a
              className="ArrowButton "
              id={undefined}
              onClick={[Function]}
              onKeyDown={[Function]}
              tabIndex="0"
            >
              <i
                alt="previous page"
                aria-hidden="true"
                className=" fa fa-chevron-left"
              />
            </a>
          </span>
          <span
            className="next-arrow"
          >
            <a
              className="ArrowButton noLeftBorder"
              id={undefined}
              onClick={[Function]}
              onKeyDown={[Function]}
              tabIndex="0"
            >
              <i
                alt="next page"
                aria-hidden="true"
                className=" fa fa-chevron-right"
              />
            </a>
          </span>
        </div>
      </div>
      <div
        className="itemContainer"
      >
        <div>
          <div
            className="cardCollection"
          >
            <div
              className="cardRow"
              id="card-row-1-0"
            >
              <div
                className="fourColumn leftAligned"
                id="card-1-0"
              >
                <img
                  alt="How to Write a Business Plan"
                  className="itemImage"
                  id="image-1-0"
                  src="/sites/default/files/2018-01/2-plan_500.jpg"
                />
                <p
                  className="itemTitle"
                  id="title-1-0"
                >
                  How to Write a Business Plan
                </p>
                <div>
                  <hr
                    className="dash"
                    id="hr-1-0"
                    style={
                      Object {
                        "width": "3.333rem",
                      }
                    }
                  />
                  <p
                    className="itemSubTitle"
                    id="subtitle-text-1-0"
                  >
                    Learn the importance of business planning, the components of a business plan, and see sample plans and resources.
                  </p>
                </div>
              </div>
              <div
                className="fourColumn leftAligned"
                id="card-1-1"
              >
                <img
                  alt="How to Write a Business Plan"
                  className="itemImage"
                  id="image-1-1"
                  src="/sites/default/files/2018-01/2-plan_500.jpg"
                />
                <p
                  className="itemTitle"
                  id="title-1-1"
                >
                  How to Write a Business Plan
                </p>
                <div>
                  <hr
                    className="dash"
                    id="hr-1-1"
                    style={
                      Object {
                        "width": "3.333rem",
                      }
                    }
                  />
                  <p
                    className="itemSubTitle"
                    id="subtitle-text-1-1"
                  >
                    Learn the importance of business planning, the components of a business plan, and see sample plans and resources.
                  </p>
                </div>
              </div>
              <div
                className="fourColumn leftAligned"
                id="card-1-2"
              >
                <img
                  alt="How to Write a Business Plan"
                  className="itemImage"
                  id="image-1-2"
                  src="/sites/default/files/2018-01/2-plan_500.jpg"
                />
                <p
                  className="itemTitle"
                  id="title-1-2"
                >
                  How to Write a Business Plan
                </p>
                <div>
                  <hr
                    className="dash"
                    id="hr-1-2"
                    style={
                      Object {
                        "width": "3.333rem",
                      }
                    }
                  />
                  <p
                    className="itemSubTitle"
                    id="subtitle-text-1-2"
                  >
                    Learn the importance of business planning, the components of a business plan, and see sample plans and resources.
                  </p>
                </div>
              </div>
              <div
                className="fourColumn leftAligned"
                id="card-1-3"
              >
                <img
                  alt="How to Write a Business Plan"
                  className="itemImage"
                  id="image-1-3"
                  src="/sites/default/files/2018-01/2-plan_500.jpg"
                />
                <p
                  className="itemTitle"
                  id="title-1-3"
                >
                  How to Write a Business Plan
                </p>
                <div>
                  <hr
                    className="dash"
                    id="hr-1-3"
                    style={
                      Object {
                        "width": "3.333rem",
                      }
                    }
                  />
                  <p
                    className="itemSubTitle"
                    id="subtitle-text-1-3"
                  >
                    Learn the importance of business planning, the components of a business plan, and see sample plans and resources.
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            className="cardCollection"
          >
            <div
              className="cardRow"
              id="card-row-2-0"
            >
              <div
                className="fourColumn leftAligned"
                id="card-2-0"
              >
                <img
                  alt="How to Write a Business Plan"
                  className="itemImage"
                  id="image-2-0"
                  src="/sites/default/files/2018-01/2-plan_500.jpg"
                />
                <p
                  className="itemTitle"
                  id="title-2-0"
                >
                  How to Write a Business Plan
                </p>
                <div>
                  <hr
                    className="dash"
                    id="hr-2-0"
                    style={
                      Object {
                        "width": "3.333rem",
                      }
                    }
                  />
                  <p
                    className="itemSubTitle"
                    id="subtitle-text-2-0"
                  >
                    Learn the importance of business planning, the components of a business plan, and see sample plans and resources.
                  </p>
                </div>
              </div>
              <div
                className="fourColumn leftAligned"
                id="card-2-1"
              >
                <img
                  alt="How to Write a Business Plan"
                  className="itemImage"
                  id="image-2-1"
                  src="/sites/default/files/2018-01/2-plan_500.jpg"
                />
                <p
                  className="itemTitle"
                  id="title-2-1"
                >
                  How to Write a Business Plan
                </p>
                <div>
                  <hr
                    className="dash"
                    id="hr-2-1"
                    style={
                      Object {
                        "width": "3.333rem",
                      }
                    }
                  />
                  <p
                    className="itemSubTitle"
                    id="subtitle-text-2-1"
                  >
                    Learn the importance of business planning, the components of a business plan, and see sample plans and resources.
                  </p>
                </div>
              </div>
              <div
                className="fourColumn leftAligned"
                id="card-2-2"
              >
                <img
                  alt="How to Write a Business Plan"
                  className="itemImage"
                  id="image-2-2"
                  src="/sites/default/files/2018-01/2-plan_500.jpg"
                />
                <p
                  className="itemTitle"
                  id="title-2-2"
                >
                  How to Write a Business Plan
                </p>
                <div>
                  <hr
                    className="dash"
                    id="hr-2-2"
                    style={
                      Object {
                        "width": "3.333rem",
                      }
                    }
                  />
                  <p
                    className="itemSubTitle"
                    id="subtitle-text-2-2"
                  >
                    Learn the importance of business planning, the components of a business plan, and see sample plans and resources.
                  </p>
                </div>
              </div>
              <div
                className="fourColumn leftAligned"
                id="card-2-3"
              >
                <img
                  alt="How to Write a Business Plan"
                  className="itemImage"
                  id="image-2-3"
                  src="/sites/default/files/2018-01/2-plan_500.jpg"
                />
                <p
                  className="itemTitle"
                  id="title-2-3"
                >
                  How to Write a Business Plan
                </p>
                <div>
                  <hr
                    className="dash"
                    id="hr-2-3"
                    style={
                      Object {
                        "width": "3.333rem",
                      }
                    }
                  />
                  <p
                    className="itemSubTitle"
                    id="subtitle-text-2-3"
                  >
                    Learn the importance of business planning, the components of a business plan, and see sample plans and resources.
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            className="cardCollection"
          >
            <div
              className="cardRow"
              id="card-row-3-0"
            >
              <div
                className="fourColumn leftAligned"
                id="card-3-0"
              >
                <img
                  alt="How to Write a Business Plan"
                  className="itemImage"
                  id="image-3-0"
                  src="/sites/default/files/2018-01/2-plan_500.jpg"
                />
                <p
                  className="itemTitle"
                  id="title-3-0"
                >
                  How to Write a Business Plan
                </p>
                <div>
                  <hr
                    className="dash"
                    id="hr-3-0"
                    style={
                      Object {
                        "width": "3.333rem",
                      }
                    }
                  />
                  <p
                    className="itemSubTitle"
                    id="subtitle-text-3-0"
                  >
                    Learn the importance of business planning, the components of a business plan, and see sample plans and resources.
                  </p>
                </div>
              </div>
              <div
                className="fourColumn leftAligned"
                id="card-3-1"
              >
                <img
                  alt="How to Write a Business Plan"
                  className="itemImage"
                  id="image-3-1"
                  src="/sites/default/files/2018-01/2-plan_500.jpg"
                />
                <p
                  className="itemTitle"
                  id="title-3-1"
                >
                  How to Write a Business Plan
                </p>
                <div>
                  <hr
                    className="dash"
                    id="hr-3-1"
                    style={
                      Object {
                        "width": "3.333rem",
                      }
                    }
                  />
                  <p
                    className="itemSubTitle"
                    id="subtitle-text-3-1"
                  >
                    Learn the importance of business planning, the components of a business plan, and see sample plans and resources.
                  </p>
                </div>
              </div>
              <div
                className="fourColumn leftAligned"
                id="card-3-2"
              >
                <img
                  alt="How to Write a Business Plan"
                  className="itemImage"
                  id="image-3-2"
                  src="/sites/default/files/2018-01/2-plan_500.jpg"
                />
                <p
                  className="itemTitle"
                  id="title-3-2"
                >
                  How to Write a Business Plan
                </p>
                <div>
                  <hr
                    className="dash"
                    id="hr-3-2"
                    style={
                      Object {
                        "width": "3.333rem",
                      }
                    }
                  />
                  <p
                    className="itemSubTitle"
                    id="subtitle-text-3-2"
                  >
                    Learn the importance of business planning, the components of a business plan, and see sample plans and resources.
                  </p>
                </div>
              </div>
              <div
                className="fourColumn leftAligned"
                id="card-3-3"
              >
                <img
                  alt="How to Write a Business Plan"
                  className="itemImage"
                  id="image-3-3"
                  src="/sites/default/files/2018-01/2-plan_500.jpg"
                />
                <p
                  className="itemTitle"
                  id="title-3-3"
                >
                  How to Write a Business Plan
                </p>
                <div>
                  <hr
                    className="dash"
                    id="hr-3-3"
                    style={
                      Object {
                        "width": "3.333rem",
                      }
                    }
                  />
                  <p
                    className="itemSubTitle"
                    id="subtitle-text-3-3"
                  >
                    Learn the importance of business planning, the components of a business plan, and see sample plans and resources.
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="paginator"
      >
        <div
          className="container paginator"
          id={undefined}
        >
          <p
            className="pagination-status"
          >
            <span>
              Showing  
            </span>
            <span
              className="bold"
            >
              1
                - 
              12
                
            </span>
            <span>
              of  
            </span>
            <span
              className="bold"
            >
              25
            </span>
          </p>
          <span
            className="previous-arrow"
          >
            <a
              className="ArrowButton "
              id={undefined}
              onClick={[Function]}
              onKeyDown={[Function]}
              tabIndex="0"
            >
              <i
                alt="previous page"
                aria-hidden="true"
                className=" fa fa-chevron-left"
              />
            </a>
          </span>
          <span
            className="next-arrow"
          >
            <a
              className="ArrowButton noLeftBorder"
              id={undefined}
              onClick={[Function]}
              onKeyDown={[Function]}
              tabIndex="0"
            >
              <i
                alt="next page"
                aria-hidden="true"
                className=" fa fa-chevron-right"
              />
            </a>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
